src/server/services/procedures/phone-iot/common.js

```diff
@@ -85,7 +85,32 @@ common.parseBool = function (val) {
     throw Error(`failed to parse bool: ${val}`);
 };
 common.parseFontSize = function (val) {
-    return types.parse.BoundedNumber(val, [0.1, 10.0]);
+    try { return types.parse.BoundedNumber(val, [0.1, 10.0]); }
```

Comment:
These should be added as custom input types rather than custom parsing handled within the service. This will ensure we have standard error messages rather than the ones used here.

Reply:
New commit moves all of these out to `input-types`, except the options parser, which I might update and generalize as a separate PR since I can change Chart to use it at the same time. I added an input type called `Enum` to implement some of the other types I was using.